src/libertem/api.py

```diff
@@ -554,7 +557,7 @@ def create_pick_analysis(self, dataset, x: int, y: int = None, z: int = None):
         parameters = {name: loc[name] for name in ['x', 'y', 'z'] if loc[name] is not None}
         return PickFrameAnalysis(dataset=dataset, parameters=parameters)
 
-    def run(self, job: Union[Job, Analysis], roi=None):
+    def run(self, job: Union[Job, Analysis], roi=None) -> np.ndarray:
```

Comment:
The return type of this method is not always `ndarray`, as noted in the docstring. If you want to express something like "either X or Y", you can use the `Union` type, see also [the official documentation](https://docs.python.org/3/library/typing.html#typing.Union). The whole page on typing should be of interest for fixing this issue